Source/WebCore/accessibility/AccessibilityRenderObject.cpp

```diff
@@ -554,7 +554,7 @@ bool AccessibilityRenderObject::isAttachment() const
     // Widgets are the replaced elements that we represent to AX as attachments
     bool isWidget = renderer->isWidget();
 
-    return isWidget && ariaRoleAttribute() == AccessibilityRole::Unknown;
+    return isWidget;
```

Comment:
whats the change that added the AccessibilityRole::Unknown check? I have a feeling that if someone buttons a button role on a widget element, we don't want to treat it like a widget

Reply:
This was present in first commit on 6th May 2008 as well: [Link](https://github.com/WebKit/WebKit/commit/ca5f21d630582a3f99fa65a6c882ec552ec10756)

Reply:
As discussed on Slack, I just undid this change.